tests/phpunit/tests/user.php

```diff
@@ -2353,4 +2353,79 @@ function ( $meta_id, $object_id, $meta_key ) use ( &$db_update_count ) {
+			'user_email' => 'testuser@example.com',
+		);
+
+		$user_id = wp_insert_user( $userdata );
+
+		// Assert that `wp_set_password` was triggered once during user creation.
+		$this->assertSame( 1, $mock_action->get_call_count(), 'wp_set_password was not triggered during user creation.' );
+		$args = $mock_action->get_args();
+		$this->assertSame( $userdata['user_pass'], $args[0][0], 'Wrong password argument in action.' );
+		$this->assertSame( $user_id, $args[0][1], 'Wrong user ID in action.' );
+
+		$mock_action->reset(); // Reset mock for the next scenario.
+
+		// Step 2: Update the user password explicitly via `wp_set_password`.
+		$updated_password = 'updatedpassword123';
+		wp_set_password( $updated_password, $user_id );
+
+		// Assert that `wp_set_password` was triggered once during the password update.
+		$this->assertSame( 1, $mock_action->get_call_count(), 'wp_set_password was not triggered during password update.' );
+		$args = $mock_action->get_args();
+		$this->assertSame( $updated_password, $args[0][0], 'Wrong password argument in action when updating user.' );
+		$this->assertSame( $user_id, $args[0][1], 'Wrong user ID in action when updating user.' );
+
+		$mock_action->reset(); // Reset mock for the final scenario.
+
+		// Step 3: Update user data without changing the password and ensure no action is triggered.
+		$non_password_userdata = array(
+			'ID'         => $user_id,
+			'first_name' => 'UpdatedName',
+			'user_login' => $userdata['user_login'], // Keep unchanged.
+		);
+
+		wp_insert_user( $non_password_userdata );
+
+		// Assert that `wp_set_password` is not triggered for non-password updates.
+		$this->assertSame( 0, $mock_action->get_call_count(), 'wp_set_password was incorrectly triggered for non-password update.' );
+	}
+
+	public function test_set_password_action_on_user_update() {
+		$mock_action = new MockAction();
+		add_action( 'wp_set_password', array( $mock_action, 'action' ), 10, 3 );
+
+		// Step 1: Create a user
+		$user_id = $this->factory()->user->create(
+			array(
+				'role'       => 'subscriber',
+				'user_login' => 'testuser_update',
+				'user_email' => 'testuser_update@example.com',
+				'user_pass'  => 'initialpassword',
+			)
+		);
+
+		// Assert initial password action (creation)
+		$this->assertSame( 1, $mock_action->get_call_count(), 'wp_set_password was not triggered during user creation.' );
+		$mock_action->reset(); // Reset for update test
+
+		// Step 2: Update the password via wp_set_password
+		$updated_password = 'newpassword123';
+		wp_set_password( $updated_password, $user_id );
+
+		// Assert action triggered
+		$this->assertSame( 1, $mock_action->get_call_count(), 'wp_set_password was not triggered during password update.' );
+		$args = $mock_action->get_args();
+		$this->assertSame( $updated_password, $args[0][0], 'Invalid password in wp_set_password action.' );
+		$this->assertSame( $user_id, $args[0][1], 'Invalid user ID in wp_set_password action.' );
+	}
```

Comment:
```suggestion
```
Actually, this whole test is a dupe of `Tests_Auth::test_wp_set_password_action()`